controllers/flp/fmstatus/flowmetrics_status.go

```diff
@@ -0,0 +1,113 @@
+
+func Reset() {
+	mapStatuses = make(map[types.NamespacedName]*metav1.Condition)
+	mapCards = make(map[types.NamespacedName]*metav1.Condition)
+}
+
+func SetPending(fm *metricslatest.FlowMetric) {
+	nsname := types.NamespacedName{Name: fm.Name, Namespace: fm.Namespace}
+	mapStatuses[nsname] = &metav1.Condition{
+		Type:    "Ready",
+		Reason:  "Ready",
+		Message: "flowlogs-pipeline not yet configured",
+		Status:  metav1.ConditionFalse,
+	}
+}
+
+func SetReady(fm *metricslatest.FlowMetric) {
+	nsname := types.NamespacedName{Name: fm.Name, Namespace: fm.Namespace}
+	mapStatuses[nsname] = &metav1.Condition{
+		Type:    "Ready",
+		Reason:  "Ready",
+		Message: "flowlogs-pipeline configured",
+		Status:  metav1.ConditionTrue,
+	}
+}
+
+func SetFailure(fm *metricslatest.FlowMetric, msg string) {
+	nsname := types.NamespacedName{Name: fm.Name, Namespace: fm.Namespace}
+	mapStatuses[nsname] = &metav1.Condition{
+		Type:    "Ready",
+		Reason:  "Failure",
+		Message: msg,
+		Status:  metav1.ConditionFalse,
+	}
+}
+
+func CheckCardinality(fm *metricslatest.FlowMetric) {
+	report, err := helper.CheckCardinality(fm.Spec.Labels...)
+	if err != nil {
+		SetFailure(fm, err.Error())
+		return
+	}
+	cardinality := report.GetOverall()
+	status := metav1.ConditionTrue
+	if cardinality == config.CardinalityWarnAvoid || cardinality == config.CardinalityWarnUnknown {
+		status = metav1.ConditionFalse
+	}
+	nsname := types.NamespacedName{Name: fm.Name, Namespace: fm.Namespace}
+	mapCards[nsname] = &metav1.Condition{
+		Type:    "CardinalityOK",
+		Reason:  string(cardinality),
+		Message: report.GetDetails(),
+		Status:  status,
+	}
+	SetReady(fm)
+}
+
+func Sync(ctx context.Context, c client.Client, fm *metricslatest.FlowMetricList) {
+	for i := range fm.Items {
+		nsname := types.NamespacedName{Name: fm.Items[i].Name, Namespace: fm.Items[i].Namespace}
+		// main condition is mandatory; cardinality condition is optional
+		if cond, ok := mapStatuses[nsname]; ok {
+			cardCond := mapCards[nsname]
+			setStatus(ctx, c, nsname, cond, cardCond)
+		}
+	}
+}
+
+func setStatus(ctx context.Context, c client.Client, nsname types.NamespacedName, mainCond *metav1.Condition, cardinalityCond *metav1.Condition) {
+	log := log.FromContext(ctx)
+
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		fm := metricslatest.FlowMetric{}
+		if err := c.Get(ctx, nsname, &fm); err != nil {
+			log.WithValues("NsName", nsname).Error(err, "failed to get FlowMetrics status")
+			if errors.IsNotFound(err) {
+				// ignore: when it's being deleted, there's no point trying to update its status
+				return nil
+			}
+			return err
+		}
+		if mainCond != nil {
+			meta.SetStatusCondition(&fm.Status.Conditions, *mainCond)
+		}
+		if cardinalityCond != nil {
+			meta.SetStatusCondition(&fm.Status.Conditions, *cardinalityCond)
+		}
+		return c.Status().Update(ctx, &fm)
+	})
+
+	if err != nil {
+		log.Error(err, "failed to update FlowMetrics status")
+	}
+}
```

Comment:
it might be a good idea adding UT for this while u are here

Reply:
done